src/main/scala/is/hail/io/vcf/LoadVCF.scala

```diff
@@ -296,21 +303,57 @@ object LoadVCF {
               reportAcc += VCFReport.Symbolic
               None
             } else
-              Some(reader.readRecord(reportAcc, vc, infoSignatureBc.map(_.value),
-                settings))
+              Some(reader.readRecord(codec, reportAcc, vc, infoSignatureBc.map(_.value), genotypeSignatureBc.value, settings))
           }
         }.value
         }
       }.toOrderedRDD(justVariants)
 
     justVariants.unpersist()
 
-    VariantSampleMatrix(hc, VariantMetadata(sampleIds,
-      Annotation.emptyIndexedSeq(sampleIds.length),
+    LoadVCFResult(rdd = rdd, sampleIds = sampleIds, vaSignature = variantAnnotationSignatures, gSignature = genotypeSignature)
+  }
+
+  def apply(hc: HailContext,
+    file1: String,
+    files: Array[String], // FIXME hack
+    storeGQ: Boolean,
+    nPartitions: Option[Int],
+    skipGenotypes: Boolean,
+    ppAsPL: Boolean,
+    skipBadAD: Boolean): VariantDataset = {
+
+    val settings = VCFSettings(storeGQ, skipGenotypes, ppAsPL, skipBadAD)
+    val reader = new GenotypeRecordReader()
+
+    val result = LoadVCF(hc, reader, file1, files, nPartitions, skipGenotypes, settings, genericGenotypes = false)
+
+    VariantSampleMatrix(hc, VariantMetadata(result.sampleIds,
+      Annotation.emptyIndexedSeq(result.sampleIds.length),
       Annotation.empty,
       TStruct.empty,
-      variantAnnotationSignatures,
-      TStruct.empty), genotypes)
+      result.vaSignature,
+      TStruct.empty,
+      TGenotype,
+      isGenericGenotype = false), result.rdd)
   }
 
+  def apply(hc: HailContext,
```

Comment:
I feel like these two apply methods should be able to be unified (and maybe deleted and replaced to a call to LoadVCF.apply that now returns a LoadVCFResult).  I think the caller can create the RecordReader and type-specific args can go into the RecordReaders.